shell/common/shorebird/shorebird.cc

```diff
@@ -79,6 +79,119 @@ FileCallbacks ShorebirdFileCallbacks() {
+  // lookups will fail.  Change your Scheme to Release to fix:
+  // https://github.com/flutter/flutter/wiki/Debugging-the-engine#debugging-ios-builds-with-xcode
+  // FML_CHECK(DartSnapshot::VMSnapshotFromSettings(settings))
+  //     << "XCode Scheme must be set to Release to use Shorebird";
+
+  auto shorebird_updater_dir_name = "shorebird_updater";
+
+  auto code_cache_dir = fml::paths::JoinPaths(
+      {std::move(args.code_cache_path), shorebird_updater_dir_name});
+  auto app_storage_dir = fml::paths::JoinPaths(
+      {std::move(args.app_storage_path), shorebird_updater_dir_name});
+
+  fml::CreateDirectory(fml::paths::GetCachesDirectory(),
+                       {shorebird_updater_dir_name},
+                       fml::FilePermission::kReadWrite);
+
+  bool init_result;
+  // Using a block to make AppParameters lifetime explicit.
+  {
+    FML_LOG(INFO) << "Constructing app parameters";
+    AppParameters app_parameters;
+    // Combine version and version_code into a single string.
+    // We could also pass these separately through to the updater if needed.
+    auto release_version =
+        args.release_version.version + "+" + args.release_version.build_number;
+    app_parameters.release_version = release_version.c_str();
+    app_parameters.code_cache_dir = code_cache_dir.c_str();
+    app_parameters.app_storage_dir = app_storage_dir.c_str();
+
+    // https://stackoverflow.com/questions/26032039/convert-vectorstring-into-char-c
+    std::vector<const char*> c_paths{};
+    c_paths.push_back(args.release_app_library_path.c_str());
+    // for (const auto& string : settings.application_library_path) {
+    //   FML_LOG(INFO) << "Pushing " << string << " to c_paths";
+    //   c_paths.push_back(string.c_str());
+    // }
+    // Do not modify application_library_path or c_strings will invalidate.
+
+    app_parameters.original_libapp_paths = c_paths.data();
+    app_parameters.original_libapp_paths_size = c_paths.size();
+
+    FML_LOG(INFO) << "Calling shorebird_init";
+    // shorebird_init copies from app_parameters and shorebirdYaml.
+    init_result = shorebird_init(&app_parameters, ShorebirdFileCallbacks(),
+                                 args.shorebird_yaml.c_str());
+    FML_LOG(INFO) << "init_result: " << init_result;
+  }
+
+  // We've decided not to support synchronous updates on launch for now.
+  // It's a terrible user experience (having the app hang on launch) and
+  // instead we will provide examples of how to build a custom update UI
+  // within Dart, including updating as part of login, etc.
+  // https://github.com/shorebirdtech/shorebird/issues/950
+
+  // We only set the base snapshot on iOS for now.
+  // TODO: this won't compile as we don't have a settings object here.
+  // #if FML_OS_IOS || FML_OS_MACOSX
+  //   SetBaseSnapshot(settings);
+  // #endif
+
+  FML_LOG(INFO) << "Checking for active patch";
+  char* c_active_path = shorebird_next_boot_patch_path();
+  std::string active_path = "";
```

Comment:
I think you don't need the `= ""`, just the ;

Reply:
Yeah, this function needs to be consolidated with the other ConfigureShorebird. Going to do that in a followup PR as it will affect the macOS bits